### ISSUE TITLE
[Bug]: Passing Location objects into createMemoryRouter's initialEntries doesn't type-check

### ISSUE THREAD

**Issue by ShaneDrury** [remix-run/react-router#9470]
### What version of React Router are you using?

6.4.2

### Steps to Reproduce

- Use a `Partial<Location>` object in `initialEntries` when creating a memory router:
```tsx
import { createMemoryRouter } from 'react-router-dom';
const router = createMemoryRouter([], {
  initialEntries: [{ pathname: '/foo', state: { myState: 'bar' } }],
});
```

### Expected Behavior

The code type-checks and works as normal, where the state is accessible in the components with `useLocation`.

### Actual Behavior

The code does not type-check but works as normal, where the state is accessible in the components with `useLocation`.
The type error is:
```
TS2322: Type '{ pathname: string; state: { myState: string; }; }' is not assignable to type 'string'.
```

This seems to be because [initialEntries](https://github.com/remix-run/react-router/blob/5f3cfb7ac2ad823ecdf69069332567e35ad795c6/packages/react-router/index.ts#L206) of [createMemoryRouter](https://github.com/remix-run/react-router/blob/5f3cfb7ac2ad823ecdf69069332567e35ad795c6/packages/react-router/index.ts#L201) is typed as `string[]` rather than `InitialEntry[]` which is what [createMemoryHistory](https://github.com/remix-run/react-router/blob/5f3cfb7ac2ad823ecdf69069332567e35ad795c6/packages/router/history.ts#L205) uses and `initialEntries` is passed on to.

**Comment by ShaneDrury** [comment]
The [documentation](https://reactrouter.com/en/main/routers/create-memory-router#type-declaration) also seems to show the type as `InitialEntry[]`.

**Comment by brophdawg11** [comment]
Thanks!  Just merged a fix for this in #9498, should be out in a `6.4.3-pre.1` hopefully today and then a stable 6.4.3 later this week or next

**Comment by ShaneDrury** [comment]
Thanks a lot @brophdawg11! I appreciate it 👍
